### DISCUSSION TITLE
Support for configuration files located at `.renovate/renovate.json[5]`

### DISCUSSION THREAD

**Discussion by dalbani** [renovatebot/renovate#22931] [Suggest an Idea]
### Tell us more.

Hello,

Here's some context: my goal to share a common Renovate Bot configuration over multiple Git projects.
I do have read the documentation at https://docs.renovatebot.com/config-presets/, but my understanding is that using such a preset requires specifiying it in renovate.json (or any supported variation thereof).
While it's not a lot of text, there's always the change of someone making a typo when setting up a new project in our organization.
To work around that possibility, I was thinking of using a Git submodule pointing to a repository containing the necessary configuration referencing the preset.
Unfortunality, given that Git submodules work at the directory level, the only choices at the moment are `.github/` and `.gitlab/`, according to https://docs.renovatebot.com/configuration-options/.
But I'm already using them to store files — actually, I'm using only one, but that would seem confusing to put files in a directory named after a CI environment that I don't use.
My proposal is thus to support `.renovate/` as well.

What do you think?

**Comment by rarkins** [comment]
Preset configs can be any file name at all, with the default being default.json

**Reply by dalbani** [reply]
But where in my project do I reference this preset file then?

**Reply by dalbani** [reply]
My original point was than I don't want to copy/paste the same snippet of Renovate Bot configuration in all my projects.
At least not manually, I'd prefer to rely on Git submodule which contains a "curated" configuration.

**Reply by rarkins** [reply]
Are you self hosting renovate? If so then you can set global config once with requireConfig=optional so no repo config is mandatory

**Reply by dalbani** [reply]
I do use self hosting, but I don't want either to force *all* repositories to have the same configuration.
It has to be opt-in, via a Git submodule in that case.
And I suppose that other people _not_ self hosting would have the same need as I do.

**Reply by rarkins** [reply]
Then they add a one like config to their repo extending your central one. This is a common usage, does not require any new file names

**Reply by viceice** [reply]
btw: renovate doesn't support reading repo config from git submodule.

**Reply by dalbani** [reply]
Thanks @viceice for the tip, then the case is closed 😄 
I'll follow your suggestion, @rarkins 👍

**Comment by HonkingGoose** [comment]
Heads-up we may remove JSON5 support completely in a upcoming major release (version 39). See the top item in this issue:

- #25608
